### PR TITLE
Code Serialization + Formatter 

### PR DESCRIPTION
**Pull request by warfaj**
All changes required for manifest -> code (including formatter)
Associated test included that takes a pax file + manifest and checks serialization + format diff

Big things this enables:

- A function that takes ComponentDefinition in the Manifest -> Pax Code
- A function that takes a string of Pax and returns a formatted string of Pax
- Support for comments within the language that will be honored through serialization + formatting
- A standalone pax-cli fmt that supports both .pax files and .rs files

In order to support comments within Pax, I had to modify both the grammar and the manifest to support them.
Note: we only support a single top level comment in an expression. This is mostly because parsing logic would have gotten a lot more complex to support interspersed optional comments at any point in an expr.

### CODE REVIEW

File: pax-compiler/src/expressions.rs [hunk 1]
```diff
@@ -163,121 +183,78 @@ fn merge_inline_settings_with_settings_block(
...
+
+                    let mut whitespace_removed_input = input.clone().token_value;
+                    whitespace_removed_input.retain(|c| !c.is_whitespace());
 
+                    let source_map_id = source_map.insert(input.clone());
+                    let input_statement =
+                        source_map.generate_mapped_string(whitespace_removed_input, source_map_id);
+
+                    if let Some(l) = input.token_location.clone() {
+                        if l.start_line_col.0 == 109 {
```
Comment (zackbrown) [inline pax-compiler/src/expressions.rs:254]: Vestigial debug statement?

File: pax-compiler/src/formatting/rules.rs [hunk 2]
```diff
@@ -0,0 +1,938 @@
...
+impl Child {
+    fn new(node_type: Rule, formatted_node: String) -> Self {
+        Self {
+            node_type,
+            formatted_node,
+        }
+    }
+}
+
+trait FormattingRule {
```
Comment (zackbrown) [inline pax-compiler/src/formatting/rules.rs:207]: 👍  on design ✨

**Review by zackbrown — CHANGES_REQUESTED** [review]
Beautiful!  Meaty!  One small request for change, then LGTM

File: pax-compiler/src/expressions.rs (hunk 1 above)
Comment (warfaj) [reply pax-compiler/src/expressions.rs]: Removed

**Pull request merged**